nemo/collections/asr/data/audio_to_text_lhotse_target_language.py

```diff
@@ -0,0 +1,243 @@
+            'a_sig_length': NeuralType(tuple('B'), LengthsType()),
+            'transcripts': NeuralType(('B', 'T'), LabelsType()),
+            'transcript_length': NeuralType(tuple('B'), LengthsType()),
+            'sample_id': NeuralType(tuple('B'), LengthsType(), optional=True),
+            'target_lang_id': NeuralType(('B', 'T', 'D'), LabelsType()),
+        }
+
+    def __init__(self, tokenizer, cfg):
+        super().__init__()
+        self.tokenizer = TokenizerWrapper(tokenizer)
+        self.load_audio = AudioSamples(fault_tolerant=True)
+        self.cfg = cfg
+        self.num_languages = cfg.get('num_languages', 128)
+        self.num_sample_per_mel_frame = cfg.get('num_sample_per_mel_frame', 160)
+        self.num_mel_frame_per_asr_frame = cfg.get('num_mel_frame_per_asr_frame', 8)
+
+    def _get_language_index(self, language_code: str) -> int:
+        """
+        Maps language codes to indices .
+        """
+        if language_code not in self._GLOBAL_LANG_MAP:
+            raise ValueError(
+                f"Unknown language code: {language_code}. Supported languages: {list(self._GLOBAL_LANG_MAP.keys())}"
+            )
+
+        index = self._GLOBAL_LANG_MAP[language_code]
+
+        return index
+
+    def lang_to_target_lang(
+        self, cut, num_languages: int, num_sample_per_mel_frame: int, num_mel_frame_per_asr_frame: int
+    ):
+        """
+        Create language target tensor for the sequence.
+        Similar to speaker_to_target but for languages.
+        """
+        # Calculate encoder output length
+        encoder_hidden_len = self.get_hidden_length_from_sample_length(
+            cut.num_samples, num_sample_per_mel_frame, num_mel_frame_per_asr_frame
+        )
+
+        # Initialize language target matrix
+        mask = np.zeros((num_languages, encoder_hidden_len))
+
+        # Get language index
+        lang_id = self._get_language_index(cut.supervisions[0].language)
+        # Set the corresponding language ID to 1 for all time steps
+        mask[lang_id, :] = 1
+
+        return mask
+
+    # this function is taken from nemo.collections.asr.parts.utils.asr_multispeaker_utils import get_hidden_length_from_sample_length
+    def get_hidden_length_from_sample_length(
+        self, num_samples: int, num_sample_per_mel_frame: int = 160, num_mel_frame_per_asr_frame: int = 8
+    ) -> int:
+        """
+        Calculate the hidden length from the given number of samples.
+
+        This function computes the number of frames required for a given number of audio samples,
+        considering the number of samples per mel frame and the number of mel frames per ASR frame.
+
+        Parameters:
+            num_samples (int): The total number of audio samples.
+            num_sample_per_mel_frame (int, optional): The number of samples per mel frame. Default is 160.
+            num_mel_frame_per_asr_frame (int, optional): The number of mel frames per ASR frame. Default is 8.
```

Comment:
Better to rename these to shift and subsampling factor.

Reply:
https://github.com/NVIDIA/NeMo/blob/92f2defd7686866a470e5e1c51f44b85f31e6326/examples/asr/conf/fastconformer/fast-conformer_transducer_bpe.yaml#L101